items:
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  id: _adls_error_exception
  parent: com.microsoft.azure.management.datalake.store.models
  children:
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a49aecd24fddc92ff7942444e98b6f062
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a700f96f5cac52a597e0a13be6afa4995
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a214d27890b2eca72fd0eb5f8692baab8
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a6c1326f1e3d11199ff61cc6cfba30ec6
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a8bc2d576657cbc0ec69c923d27a3d902
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a6610888d7332dcd58f622e25a125ed50
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a132a55ea572e5e6149dbdcb9e76cf888
  - com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a7777bc4241c58bc9f6cbf4d8eba23cde
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: AdlsErrorException
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException
  type: Class
  source:
    remote: &o0
      path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
      branch: master
      repo: https://github.com/Azure/azure-sdk-for-java.git
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 14
  package: com.microsoft.azure.management.datalake.store.models
  summary: <p>Exception thrown for an invalid response with <xref href="com.microsoft.azure.management.datalake.store.models._adls_error" data-throw-if-not-resolved="false"></xref> information. </p>
  syntax: &o1
    content: public class AdlsErrorException extends RestException
  inheritance:
  - java.lang.Object
  - Exception
  - com.microsoft.rest._rest_exception
  inheritedMembers:
  - com.microsoft.rest._rest_exception_1a8b86502c682e0b0e47f60b1c33b67b8f
  - com.microsoft.rest._rest_exception_1a8a0b21d2d492219e29a07567ff52b9cf
  - com.microsoft.rest._rest_exception_1ac00611a5b9a02b99e257f2407dc679d5
  - com.microsoft.rest._rest_exception_1ae3378da3ede0800826ccfe5f68532460
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a49aecd24fddc92ff7942444e98b6f062
  id: _adls_error_exception_1a49aecd24fddc92ff7942444e98b6f062
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: AdlsErrorException()
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException.AdlsErrorException()
  type: Constructor
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 26
  package: com.microsoft.azure.management.datalake.store.models
  summary: <p>Initializes a new instance of the <xref href="com.microsoft.azure.management.datalake.store.models._adls_error_exception" data-throw-if-not-resolved="false"></xref> class. </p>
  syntax:
    content: public AdlsErrorException()
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a700f96f5cac52a597e0a13be6afa4995
  id: _adls_error_exception_1a700f96f5cac52a597e0a13be6afa4995
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: AdlsErrorException(final String message)
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException.AdlsErrorException(final String message)
  type: Constructor
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 32
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.management.datalake.store.models._adls_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public AdlsErrorException(final String message)
    parameters:
    - id: message
      type: 3d93591e
      description: <p>The exception message. </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a214d27890b2eca72fd0eb5f8692baab8
  id: _adls_error_exception_1a214d27890b2eca72fd0eb5f8692baab8
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: AdlsErrorException(final String message, final Throwable cause)
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException.AdlsErrorException(final String message, final Throwable cause)
  type: Constructor
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 41
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.management.datalake.store.models._adls_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public AdlsErrorException(final String message, final Throwable cause)
    parameters:
    - id: message
      type: 3d93591e
      description: <p>the exception message </p>
    - id: cause
      type: fa00b19f
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a6c1326f1e3d11199ff61cc6cfba30ec6
  id: _adls_error_exception_1a6c1326f1e3d11199ff61cc6cfba30ec6
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: AdlsErrorException(final Throwable cause)
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException.AdlsErrorException(final Throwable cause)
  type: Constructor
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 49
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Initializes a new instance of the <xref href="com.microsoft.azure.management.datalake.store.models._adls_error_exception" data-throw-if-not-resolved="false"></xref> class.</p>

    <p></p>
  syntax:
    content: public AdlsErrorException(final Throwable cause)
    parameters:
    - id: cause
      type: fa00b19f
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a8bc2d576657cbc0ec69c923d27a3d902
  id: _adls_error_exception_1a8bc2d576657cbc0ec69c923d27a3d902
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: getBody()
  fullName: AdlsError com.microsoft.azure.management.datalake.store.models.AdlsErrorException.getBody()
  type: Method
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 65
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Gets the HTTP response body.</p>

    <p></p>
  syntax:
    content: public AdlsError getBody()
    return:
      type: com.microsoft.azure.management.datalake.store.models._adls_error
      description: <p>the response body </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a6610888d7332dcd58f622e25a125ed50
  id: _adls_error_exception_1a6610888d7332dcd58f622e25a125ed50
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: getResponse()
  fullName: Response com.microsoft.azure.management.datalake.store.models.AdlsErrorException.getResponse()
  type: Method
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 57
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Gets information about the associated HTTP response.</p>

    <p></p>
  syntax:
    content: public Response getResponse()
    return:
      type: e9d34ed6
      description: <p>the HTTP response </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a132a55ea572e5e6149dbdcb9e76cf888
  id: _adls_error_exception_1a132a55ea572e5e6149dbdcb9e76cf888
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: setBody(AdlsError body)
  fullName: void com.microsoft.azure.management.datalake.store.models.AdlsErrorException.setBody(AdlsError body)
  type: Method
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 81
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Sets the HTTP response body.</p>

    <p></p>
  syntax:
    content: public void setBody(AdlsError body)
    parameters:
    - id: body
      type: com.microsoft.azure.management.datalake.store.models._adls_error
      description: <p>the response body </p>
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception_1a7777bc4241c58bc9f6cbf4d8eba23cde
  id: _adls_error_exception_1a7777bc4241c58bc9f6cbf4d8eba23cde
  parent: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  langs:
  - java
  name: setResponse(Response response)
  fullName: void com.microsoft.azure.management.datalake.store.models.AdlsErrorException.setResponse(Response response)
  type: Method
  source:
    remote: *o0
    path: azure-mgmt-datalake-store/src/main/java/com/microsoft/azure/management/datalake/store/models/AdlsErrorException.java
    startLine: 73
  package: com.microsoft.azure.management.datalake.store.models
  summary: >-
    <p>Sets the HTTP response.</p>

    <p></p>
  syntax:
    content: public void setResponse(Response response)
    parameters:
    - id: response
      type: e9d34ed6
      description: <p>the HTTP response </p>
references:
- uid: 3d93591e
  spec.java:
  - name: final String
    fullName: final String
- uid: fa00b19f
  spec.java:
  - name: final Throwable
    fullName: final Throwable
- uid: e9d34ed6
  spec.java:
  - name: Response
    fullName: Response
- uid: com.microsoft.rest._rest_exception_1a8b86502c682e0b0e47f60b1c33b67b8f
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException()
  fullName: com.microsoft.rest.RestException.RestException()
  type: Constructor
  summary: <p>Initializes a new instance of the AutoRestException class. </p>
  syntax:
    content: public RestException()
- uid: com.microsoft.rest._rest_exception_1a8a0b21d2d492219e29a07567ff52b9cf
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException(String message)
  fullName: com.microsoft.rest.RestException.RestException(String message)
  type: Constructor
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(String message)
    parameters:
    - id: message
      type: "26831127"
      description: <p>The exception message. </p>
- uid: com.microsoft.rest._rest_exception_1ac00611a5b9a02b99e257f2407dc679d5
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException(Throwable cause)
  fullName: com.microsoft.rest.RestException.RestException(Throwable cause)
  type: Constructor
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(Throwable cause)
    parameters:
    - id: cause
      type: a04fafd7
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.rest._rest_exception_1ae3378da3ede0800826ccfe5f68532460
  parent: com.microsoft.rest._rest_exception
  href: com.microsoft.rest._rest_exception.yml
  name: RestException(String message, Throwable cause)
  fullName: com.microsoft.rest.RestException.RestException(String message, Throwable cause)
  type: Constructor
  summary: >-
    <p>Initializes a new instance of the AutoRestException class.</p>

    <p></p>
  syntax:
    content: public RestException(String message, Throwable cause)
    parameters:
    - id: message
      type: "26831127"
      description: <p>the exception message </p>
    - id: cause
      type: a04fafd7
      description: <p>exception that caused this exception to occur </p>
- uid: com.microsoft.rest._rest_exception
  parent: com.microsoft.rest
  href: com.microsoft.rest._rest_exception.yml
  name: RestException
  fullName: com.microsoft.rest.RestException
  type: Class
  summary: <p>Exception thrown for an invalid response with custom error information. </p>
  syntax:
    content: public class RestException
- uid: com.microsoft.azure.management.datalake.store.models._adls_error
  parent: com.microsoft.azure.management.datalake.store.models
  href: com.microsoft.azure.management.datalake.store.models._adls_error.yml
  name: AdlsError
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsError
  type: Class
  summary: <p>Data Lake Store filesystem error containing a specific WebHDFS exception. </p>
  syntax:
    content: public class AdlsError
- uid: com.microsoft.azure.management.datalake.store.models._adls_error_exception
  parent: com.microsoft.azure.management.datalake.store.models
  href: com.microsoft.azure.management.datalake.store.models._adls_error_exception.yml
  name: AdlsErrorException
  fullName: com.microsoft.azure.management.datalake.store.models.AdlsErrorException
  type: Class
  summary: <p>Exception thrown for an invalid response with <xref href="com.microsoft.azure.management.datalake.store.models._adls_error" data-throw-if-not-resolved="false"></xref> information. </p>
  syntax: *o1
